src/main/java/com/aws/greengrass/mqtt/bridge/clients/MQTTClient.java

```diff
@@ -222,12 +217,18 @@ public void updateSubscriptions(Set<String> topics, Consumer<MqttMessage> messag
         }
     }
 
-    @SuppressWarnings("PMD.AvoidCatchingGenericException")
-    private void updateSubscriptionsInternal() {
+    private void cancelUpdateSubscriptionsTask() {
         synchronized (subscribeLock) {
             if (subscribeFuture != null) {
                 subscribeFuture.cancel(true);
             }
+        }
+    }
+
+    @SuppressWarnings("PMD.AvoidCatchingGenericException")
+    private void updateSubscriptionsInternal() {
+        synchronized (subscribeLock) {
+            cancelUpdateSubscriptionsTask();
             Set<String> topicsToRemove = new HashSet<>(subscribedLocalMqttTopics);
```

Comment:
what's up with this? Can we keep this PR targeted?

Reply:
whoops, yeah this is uncommitted stuff, didn't mean to include

Reply:
should be good now